### PR TITLE
optimize masking and remove unnecessary pad_list

### PR DESCRIPTION
**Pull request by bobchennan**
Theoretically it should improve both training time and GPU memory usage.

### CODE REVIEW

**Comment by bobchennan** [comment]
There seems to be a problem with multiple-gpu right now:

RuntimeError: The size of tensor a (179) must match the size of tensor b (178) at non-singleton dimension 1

File: espnet/nets/e2e_asr_th.py [hunk 1]
```diff
@@ -860,9 +859,9 @@ def forward(self, enc_hs_pad, enc_hs_len, dec_z, att_prev, scaling=2.0):
 
         # initialize attention weight with uniform dist.
         if att_prev is None:
-            att_prev = [enc_hs_pad.new(l).fill_(1.0 / l) for l in enc_hs_len]
             # if no bias, 0 0-pad goes 0
-            att_prev = pad_list(att_prev, 0)
+            att_prev = to_cuda(self, (1. - make_pad_mask(enc_hs_len).float()))
+            att_prev = att_prev / torch.FloatTensor(enc_hs_len).unsqueeze(-1).to(att_prev)
```
[the remark this answers is not in the document]
Comment (bobchennan) [reply espnet/nets/e2e_asr_th.py]: Yes you are right. I changed to new().

**Comment by kan-bayashi** [comment]
Is the gpu problem fixed?

**Comment by bobchennan** [comment]
I think so but I am thinking to run one exp with recent commit.

**Comment by bobchennan** [comment]
Currently I cannot get 2 gpu so if possible could you test it on wsj?

**Comment by kan-bayashi** [comment]
Now the power supply of our university is cutting off.
I will do it on Sunday.

**Comment by kan-bayashi** [comment]
I tested on WSJ. It works with ngpu = 3.

**Pull request merged**